libgraph/include/katana/GraphTopology.h

```diff
@@ -1595,6 +1598,30 @@ class KATANA_EXPORT EdgeTypeAwareBiDirTopology
   }
 };
 
+template <typename Graph>
+using has_undirected_t =
+    decltype(std::declval<Graph>().UndirectedEdges(typename Graph::Node()));
+
+template <typename Graph, typename GNode = typename Graph::Node>
+auto
+Edges(const Graph& graph, GNode node) {
+  if constexpr (katana::is_detected_v<has_undirected_t, Graph>) {
+    return graph.UndirectedEdges(node);
+  } else {
+    return graph.OutEdges(node);
+  }
+}
+
+template <typename Graph, typename GEdge>
+auto
+EdgeDst(const Graph& graph, GEdge edge) {
+  if constexpr (katana::is_detected_v<has_undirected_t, Graph>) {
+    return graph.UndirectedEdgeNeighbor(edge);
+  } else {
+    return graph.OutEdgeDst(edge);
+  }
+}
```

Comment:
Shoud there be an `EdgeSrc` method?

Reply:
Not based on what is used by analytics applications. There it's either an out-edge destination or an undirected-edge destination. We can always add it if needed!